packages/babel-helper-module-transforms/src/index.js

```diff
@@ -150,6 +155,30 @@ export function buildNamespaceInitStatements(
   return statements;
 }
 
+export function buildReexportsFromMeta(meta, metadata, loose) {
+  const reexports = [];
+  for (const [exportName, importName] of metadata.reexports) {
+    reexports.push(
+      (loose
+        ? template.statement`EXPORTS.EXPORT_NAME = NAMESPACE.IMPORT_NAME;`
+        : template`
```

Comment:
I suspect hoisting the `template.statement` / `template` calls outside of the loop will do a lot more for performance than changing between `Array.from` or `for of`